### ISSUE TITLE
Should adding a canonical alias to a room make it free to join?

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#5599]
### Description

I see (and make) the mistake of having the room invite-only despite adding an alias for people to join quite a lot. A possible solution may be to also set the room to be accessible as part of the canonical alias being added.

Some possible solutions are:
* Automatically set the room to public
* Ask inline if the user would also like to make the room public (**my preferred method**)
  * Eg: "It looks like you're adding an alias. Would you like to make this room public? [yes] [no]"
* Ask with a popup

### Version information

- **Platform**: web (in-browser)
- **Browser**: Chrome 62
- **OS**: Windows 10
- **URL**: riot.im/develop

**Comment by turt2live** [comment]
For reference, the current conversation flow is a lot like this:
* **A**: Please join #mycoolroom
* **B**: It's not accessible?
* **A**: Oops! I set it to invite only. Try now?

**Comment by lampholder** [comment]
Your preferred method is my preferred method - sensibly prompting the user to confirm their intention to make the room public sounds good to me.

"It looks like you're adding an alias" is a little bit Clippy for me 😁  - we could try "This room is currently invite only - would you like to make it public?"

**Comment by hex-m** [comment]
The relationship between "Make this room public", set a published address, "Publish this room to the public in the room directory", "Anyone can read the history" and "Anyone how know the room's link can access it" is still confusing me after months of using Element.

With knocking coming up I would suggest there should be three basic types of Rooms:
* Private: invite-only, has no identifier/link, not listed anywhere
* Public: has a public Address, history is public, free to join
* Default: has a public Address, not listed in the room directory, anyone who knows the name can knock

This is just a suggestion but the general point is that those settings are too complex.

**Comment by Mikaela** [comment]
I disagree with automatic publishing as there are currently at least two cases where it's desirable to have aliases while not allowing public joining:

* Knocking-only rooms
* Restricted rooms for members of other rooms and spaces

Those also need a method to be discoverable and I don't think they necessarily need to be hidden under spaces, especially I don't think knocking and spaces currently have a connection.
